在vue3项目中安装没问题，运行时报fs错误，搜索后好像是说浏览器环境不能直接使用fs模块操作文件系统

Node版本：v23.11.1

步骤1：项目安装
`npm i lay-excel`

步骤2：引用
`import LAY_EXCEL from 'lay-excel'`

步骤3：启动
`npm run dev`

启动成功

打开浏览器后，报错：

**X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]**

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/node-loader.mjs:16:7:
      16 │ import 'fs';
         ╵        ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/node-loader.cjs:18:8:
      18 │ require('fs');
         ╵         ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/fs.mjs:1:25:
      1 │ import { promises } from 'fs';
        ╵                          ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/fs.cjs:3:19:
      3 │ const fs = require('fs');
        ╵                    ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)
  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@antfu/install-pkg/dist/index.cjs:54:29:
      54 │ var import_node_fs = require("fs");
         ╵                              ~~~~

终止批处理操作吗(Y/N)? y
PS C:\work\cloud-platform\platform-admin-ui> npm i xlsx-js-style
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'eslint-visitor-keys@5.0.1',
npm warn EBADENGINE   required: { node: '^20.19.0 || ^22.13.0 || >=24' },
npm warn EBADENGINE   current: { node: 'v23.11.1', npm: '10.9.2' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'svg-icon-baker@1.2.0',
npm warn EBADENGINE   current: { node: 'v23.11.1', npm: '10.9.2' }
npm warn EBADENGINE }

added 15 packages in 16s

274 packages are looking for funding
  run `npm fund` for details
PS C:\work\cloud-platform\platform-admin-ui> npm run dev

> yudao-ui-admin-vue3@2025.12-snapshot dev
> vite --mode env.local

The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

  VITE v5.1.4  ready in 32712 ms

  ➜  Local:   http://localhost:80/
  ➜  Network: http://10.30.24.99:80/
  ➜  press h + enter to show help
X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/node-loader.mjs:16:7:
      16 │ import 'fs';
         ╵        ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/node-loader.cjs:18:8:
      18 │ require('fs');
         ╵         ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/fs.cjs:3:19:
      3 │ const fs = require('fs');
        ╵                    ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@iconify/utils/lib/loader/fs.mjs:1:25:
      1 │ import { promises } from 'fs';
        ╵                          ~~~~

X [ERROR] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports specified in its package.json. [plugin vite:dep-pre-bundle]

    node_modules/esbuild/lib/main.js:1374:21:
      1374 │         let result = await callback({
           ╵                      ^

    at packageEntryFailure (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48038:17)
    at resolvePackageEntry (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48035:5)
    at tryNodeResolve (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47805:20)
    at Context.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:47568:28)
    at Object.resolveId (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:50762:64)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:67768:21
    at async file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48378:34
    at async requestCallbacks.on-resolve (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1374:22)
    at async handleRequest (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:732:11)

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at setup (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:48358:19)
    at handlePlugins (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:788:5)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:68
    at new Promise (<anonymous>)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2223:27)
    at Object.context (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:2048:58)
    at prepareEsbuildOptimizerRun (file:///C:/work/cloud-platform/platform-admin-ui/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:52520:35)

  The plugin "vite:dep-pre-bundle" was triggered by this import

    node_modules/@antfu/install-pkg/dist/index.cjs:54:29:
      54 │ var import_node_fs = require("fs");
         ╵                              ~~~~

C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1651
  let error = new Error(text);
              ^

Error: Build failed with 5 errors:
node_modules/esbuild/lib/main.js:1374:21: ERROR: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports 
specified in its package.json.
node_modules/esbuild/lib/main.js:1374:21: ERROR: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports 
specified in its package.json.
node_modules/esbuild/lib/main.js:1374:21: ERROR: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports 
specified in its package.json.
node_modules/esbuild/lib/main.js:1374:21: ERROR: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports 
specified in its package.json.
node_modules/esbuild/lib/main.js:1374:21: ERROR: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "fs". The package may have incorrect main/module/exports 
specified in its package.json.
    at failureErrorWithLog (C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1651:15)
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1059:25
    at C:\work\cloud-platform\platform-admin-ui\node_modules\esbuild\lib\main.js:1527:9
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}

Node.js v23.11.1

--- a/.npmignore
+++ b/.npmignore
@@ -4,7 +4,6 @@ _book/
 demos/
 docs/
 layui/
-layui_exts/
 node_modules/
 book.json
 index.js

--- a/cjs.js
+++ b/cjs.js
@@ -1,0 +1,10 @@
+// CJS entry point for lay-excel
+// The UMD bundle (layui_exts/excel.js) sets window.LAY_EXCEL as a side effect.
+// This CJS wrapper preserves backward compatibility for Node.js/webpack users.
+// In Node.js environments, provide a minimal global object.
+// See: https://github.com/wangerzi/layui-excel/issues/45
+if (typeof window === 'undefined') {
+  global.window = {};
+}
+require('./layui_exts/excel.js');
+module.exports = window.LAY_EXCEL;

--- a/index.mjs
+++ b/index.mjs
@@ -1,0 +1,9 @@
+// ESM entry point for lay-excel
+// The UMD bundle (layui_exts/excel.js) sets window.LAY_EXCEL as a side effect.
+// This ESM wrapper allows Vite/Rollup to correctly resolve the default export.
+// See: https://github.com/wangerzi/layui-excel/issues/45
+import './layui_exts/excel.js';
+
+const LAY_EXCEL = window.LAY_EXCEL;
+export default LAY_EXCEL;
+export { LAY_EXCEL };

--- a/package.json
+++ b/package.json
@@ -2,7 +2,8 @@
   "name": "lay-excel",
   "version": "1.8.4",
   "description": "简单快捷的导出插件，导出仅需一句话",
-  "main": "src/excel.js",
+  "main": "cjs.js",
+  "module": "index.mjs",
   "scripts": {
     "test": "",
     "build": "./node_modules/.bin/webpack-cli -c webpack.config.js",
@@ -26,14 +27,13 @@
   "homepage": "https://github.com/wangerzi/layui-excel#readme",
   "dependencies": {
     "blob": "^0.1.0",
-    "cos-nodejs-sdk-v5": "2.16.0-beta.3",
     "file-saver": "^2.0.2",
-    "fs": "0.0.1-security",
     "jszip": "^3.5.0",
     "lodash": "^4.17.21",
     "npm": "^6.14.6"
   },
   "devDependencies": {
+    "cos-nodejs-sdk-v5": "2.16.0-beta.3",
     "@babel/plugin-transform-modules-commonjs": "^7.12.1",
     "babel-core": "^6.26.3",
     "babel-loader": "^7.1.5",

